Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -4065,11 +4071,21 @@ Optional<String> Element::lang() const
         return {};
     };
 
-    // If the resulting value is the empty string, then it must be interpreted as meaning that the language of the node is explicitly unknown.
-    auto maybe_lang = attempt_to_determine_lang_attribute();
-    if (!maybe_lang.has_value() || maybe_lang->is_empty())
+    if (!m_lang_value.has_value()) {
+        // If the resulting value is the empty string, then it must be interpreted as meaning that the language of the node is explicitly unknown.
+        m_lang_value = attempt_to_determine_lang_attribute().value_or(String {});
```

Comment:
If `attempt_to_determine_lang_attribute()` returns a `String` instead of an `Optional<String>`, you no longer need this `.value_or(String {})`. Then the "attempt_" part of the name can be dropped as well, since we already work through all the steps and always return a value :)